docs/guide/deployment/oauth-server.md

```diff
@@ -0,0 +1,134 @@
+openssl ecparam -name prime256v1 -genkey -noout -out $(pwd)/.tmp/privKeys/accessTokenKey.pem
+```
+
+```bash
+# See token keys for oauth-server.
+ls .tmp/privKeys 
+accessTokenKey.pem	idTokenKey.pem
+```
+
+### How to get configuration file
+A configuration template is available on [test/oauth-server/config.yaml](https://github.com/plgd-dev/cloud/blob/v2/test/oauth-server/config.yaml). 
+You can also see `config.yaml` configuration file on the `test/oauth-server` folder by downloading `git clone https://github.com/plgd-dev/cloud.git`. 
+```bash
+# See config file on the source
+cat test/oauth-server/conifg.yaml > oauth-server.yaml
+```
+
+### Edit configuration file 
+You can edit configuration file including server port, certificates, token keys and so on.
+
+See an example of address, tls and keys on the followings.
+```yaml
+...
+apis:
+  grpc:
+    address: "0.0.0.0:9088"
+    tls:
+      caPool: "/data/certs/root_ca.crt"
+      keyFile: "/data/certs/http.key"
+      certFile: "/data/certs/http.crt"
+...
+oauthSigner:
+  idTokenKeyFile: "/secret/private/idToken.key"
+  accessTokenKeyFile: "/secret/private/accessToken.key"
+  domain: "localhost:9088"
+```
+
+### Run docker image 
+You can run plgd/authorization image using certificates and configuration file on the folder you made certificates.
+```bash
+docker run -d --network=host \
+	--name=authorization \
+	-v $(pwd)/.tmp/certs:/data/certs \
+	-v $(pwd)/.tmp/privKeys:/secret/private \
+	-v $(pwd)/oauth-server.yaml:/data/oauth-server.yaml \
+	plgd/oauth-server:latest --config=/data/oauth-server.yaml
+```
+
+## YAML Configuration
+### Logging
+
+| Property | Type | Description | Default |
+| ---------- | -------- | -------------- | ------- |
+| `log.debug` | bool | `Set to true if you would like to see extra information on logs.` | `false` |
+
+### HTTP API
+HTTP API of the OAuth Server service as defined [here](https://github.com/plgd-dev/cloud/blob/v2/test/oauth-server/uri/uri.go)
+
+| Property | Type | Description | Default |
+| ---------- | -------- | -------------- | ------- |
+| `api.http.address` | string | `Listen specification <host>:<port> for http client connection.` | `"0.0.0.0:9100"` |
+| `api.http.tls.caPool` | string | `File path to the root certificate in PEM format which might contain multiple certificates in a single file.` |  `""` |
+| `api.http.tls.keyFile` | string | `File path to private key in PEM format.` | `""` |
+| `api.http.tls.certFile` | string | `File path to certificate in PEM format.` | `""` |
+| `api.http.tls.clientCertificateRequired` | bool | `If true, require client certificate.` | `true` |
+
+### OAuth Signer
+Signer configuration to issue ID/access tokens of OAuth provider for mock testing.
+
+| Property | Type | Description | Default |
+| ---------- | -------- | -------------- | ------- |
+| `oauthSigner.idTokenKeyFile` | string | `File path to private key for ID token in PEM format.` | `""` |
```

Comment:
@jkralik please specify cert type, @Trishia please add it to the docu. Same for the access token key.

Reply:
idTokenKeyFile - must be asymtriec RSA key because UI supports only RSA. You can generate it via command:
```bash
openssl genrsa -out idToken.key 4096
```